### PR TITLE
Week 4 (Owings Mill)

### PR DESCRIPTION
**Pull request by Shom770**
This change is [Reviewable](https://reviewable.io/reviews/team4099/Crescendo-2024/41)

### CODE REVIEW

File: src/main/kotlin/com/team4099/robot2023/Robot.kt [hunk 1]
```diff
@@ -125,15 +126,15 @@ object Robot : LoggedRobot() {
 
     // Set the scheduler to log events for command initialize, interrupt, finish
     CommandScheduler.getInstance().onCommandInitialize { command: Command ->
-      Logger.recordOutput("/ActiveCommands/${command.name}", true)
```
Comment (sswadkar) [inline src/main/kotlin/com/team4099/robot2023/Robot.kt:128]: Keep

File: src/main/kotlin/com/team4099/robot2023/Robot.kt [hunk 2]
```diff
@@ -125,15 +126,15 @@ object Robot : LoggedRobot() {
...
+      DebugLogger.recordDebugOutput("/ActiveCommands/${command.name}", true)
     }
 
     CommandScheduler.getInstance().onCommandFinish { command: Command ->
-      Logger.recordOutput("/ActiveCommands/${command.name}", false)
```
Comment (sswadkar) [inline src/main/kotlin/com/team4099/robot2023/Robot.kt:132]: Keep

File: src/main/kotlin/com/team4099/robot2023/Robot.kt [hunk 3]
```diff
@@ -125,15 +126,15 @@ object Robot : LoggedRobot() {
...
+      DebugLogger.recordDebugOutput("/ActiveCommands/${command.name}", false)
     }
 
     CommandScheduler.getInstance().onCommandInterrupt { command: Command ->
-      Logger.recordOutput("/ActiveCommands/${command.name}", false)
```
Comment (sswadkar) [inline src/main/kotlin/com/team4099/robot2023/Robot.kt:136]: Keep

File: src/main/kotlin/com/team4099/robot2023/Robot.kt [hunk 4]
```diff
@@ -178,18 +179,18 @@ object Robot : LoggedRobot() {
     // motor checker stuff
     val motorCheckerStartTime = Clock.realTimestamp
     MotorChecker.periodic()
-    Logger.recordOutput(
```
Comment (sswadkar) [inline src/main/kotlin/com/team4099/robot2023/Robot.kt]: Keep all subsystem loop time stuff

**Review by sswadkar — DISMISSED** [review]

**Review by plusparth — APPROVED** [review]
> Reviewed 20 of 63 files at r1, 83 of 83 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/team4099/Crescendo-2024/41)* status: all files reviewed, 9 unresolved discussions (waiting on @Shom770 and @sswadkar)

___
*[`src/main/java/com/team4099/utils/PolynomialRegression.java` line 0 at r2](https://reviewable.io/reviews/team4099/Crescendo-2024/41#-NuH49Cp2Un-0Es1v5IP:-NuH49Cp2Un-0Es1v5IQ:b-p3lkx3) ([raw file](https://github.com/team4099/Crescendo-2024/blob/65cf20b19d573c6c32ce4630ad78f05111470400/src/main/java/com/team4099/utils/PolynomialRegression.java#L0)):*
Is there no WPILib version of this?
___
*[`src/main/kotlin/com/team4099/robot2023/Robot.kt` line 168 at r2](https://reviewable.io/reviews/team4099/Crescendo-2024/41#-NuH4NfV9a-HmpZ_8sXN:-NuH4NfV9a-HmpZ_8sXO:b-tlzmde) ([raw file](https://github.com/team4099/Crescendo-2024/blob/65cf20b19d573c6c32ce4630ad78f05111470400/src/main/kotlin/com/team4099/robot2023/Robot.kt#L168)):*
> ```Kotlin
>     //      }
>     //
>     //    allianceSelected.set(NetworkTableValue.makeString(currentAlliance))
> ```

remove these if we are not using, otherwise uncomment
___
*[`src/main/kotlin/com/team4099/robot2023/RobotContainer.kt` line 69 at r2](https://reviewable.io/reviews/team4099/Crescendo-2024/41#-NuH4gc786gFrGwjfmIA:-NuH4gc786gFrGwjfmIB:b-kreoz0) ([raw file](https://github.com/team4099/Crescendo-2024/blob/65cf20b19d573c6c32ce4630ad78f05111470400/src/main/kotlin/com/team4099/robot2023/RobotContainer.kt#L69)):*
> ```Kotlin
>   var setAmpAngle = 270.0.degrees
>   var climbAngle: () -> Angle = { setClimbAngle }
>   var ampAngle: () -> Angle = { setAmpAngle }
> ```

what do these things do?
___
*[`src/main/kotlin/com/team4099/robot2023/RobotContainer.kt` line 238 at r2](https://reviewable.io/reviews/team4099/Crescendo-2024/41#-NuH4qGZ7ADE9POzFVYw:-NuH4qGZ7ADE9POzFVYx:b-xw9r43) ([raw file](https://github.com/team4099/Crescendo-2024/blob/65cf20b19d573c6c32ce4630ad78f05111470400/src/main/kotlin/com/team4099/robot2023/RobotContainer.kt#L238)):*
> ```Kotlin
>           if (DriverStation.getAlliance().isPresent &&
>             DriverStation.getAlliance().get() == DriverStation.Alliance.Red
>           )
> ```

add utility function for `DriverStation.isDefinitelyRed()` that checks this (probably also one for `isDefinitelyBlue()`
___
*[`src/main/kotlin/com/team4099/robot2023/commands/drivetrain/TargetSpeakerCommand.kt` line 135 at r2](https://reviewable.io/reviews/team4099/Crescendo-2024/41#-NuH62TH-ZVh9cchTx4j:-NuH62TH-ZVh9cchTx4k:b1b065o) ([raw file](https://github.com/team4099/Crescendo-2024/blob/65cf20b19d573c6c32ce4630ad78f05111470400/src/main/kotlin/com/team4099/robot2023/commands/drivetrain/TargetSpeakerCommand.kt#L135)):*
> ```Kotlin
> 
>     desiredAngle = atan2(robotTSpeaker.y.inMeters - (drivetrain.robotVelocity.y * robotTSpeaker.translation.magnitude.absoluteValue / 5).value, robotTSpeaker.x.inMeters - (drivetrain.robotVelocity.x * robotTSpeaker.translation.magnitude.absoluteValue / 5).value).radians
>     val filteredDesiredAngle = angleMedianFilter.calculate(desiredAngle.inDegrees)
> ```

yo what is going on here why are we dividing by 5?

also, I am not a believer in the median filter being given a robot relative angle. this is a recipe for missing shots when you are moving and I have no idea how this is working. you should LPF `odomTSpeaker` rather than do this

**Pull request merged**